request.go

```diff
@@ -1663,6 +1837,11 @@ func (r *Request) Expect() *Response {
 	r.chain.enter("Expect()")
 	defer r.chain.leave()
 
+	if r.hasExpected {
+		r.chain.fail(r.unexpectedExpectError("Expect"))
+		return nil
```

Comment:
Expect should never return nil. On error, it should return non-nil, but failed Response. Like this:

```
		return newResponse(responseOpts{
			config: r.config,
			chain:  r.chain,
		})
```